dipy/align/imaffine.py

```diff
@@ -0,0 +1,1010 @@
+from .imwarp import (get_direction_and_spacings, ScaleSpace)
+from .scalespace import IsotropicScaleSpace
+
+
+class AffineMap(object):
+    def __init__(self, affine, domain_grid_shape=None, domain_grid2world=None,
+                 codomain_grid_shape=None, codomain_grid2world=None):
+        """ AffineMap
+
+        Implements an affine transformation whose domain is given by
+        `domain_grid` and `domain_grid2world`, and whose co-domain is
+        given by `codomain_grid` and `codomain_grid2world`. This domain
+        and co-domain information is used as default sampling information
+        to transform images. If such domain/co-domain information is not
+        provided (None) then the sampling information needs to be specified
+        each time the `transform` or `transform_inverse` is called to
+        transform images. Note that such sampling information is not
+        necessary to transform points defined in physical space, such as
+        stream lines.
+
+        Parameters
+        ----------
+        affine : array, shape (dim + 1, dim + 1)
+            the matrix defining the affine transform, where `dim` is the
+            dimension of the space this map operates in (2 for 2D images,
+            3 for 3D images)
+        domain_grid_shape : sequence, shape (dim,), optional
+            the shape of the default domain sampling grid. When `transform`
+            is called to transform an image, the resulting image will have
+            this shape, unless a different sampling information is provided.
+            If None, then the sampling grid shape must be specified each time
+            the `transform` method is called.
+        domain_grid2world : array, shape (dim + 1, dim + 1), optional
+            the grid-to-world transform associated with the domain grid.
+            If None (the default), then the grid-to-world transform is assumed
+            to be the identity.
+        codomain_grid_shape : sequence of integers, shape (dim,)
+            the shape of the default co-domain sampling grid. When
+            `transform_inverse` is called to transform an image, the resulting
+            image will have this shape, unless a different sampling
+            information is provided. If None (the default), then the sampling
+            grid shape must be specified each time the `transform_inverse`
+            method is called.
+        codomain_grid2world : array, shape (dim + 1, dim + 1)
+            the grid-to-world transform associated with the co-domain grid.
+            If None (the default), then the grid-to-world transform is assumed
+            to be the identity.
+        """
+        self.set_affine(affine)
+        self.domain_shape = domain_grid_shape
+        self.domain_grid2world = domain_grid2world
+        self.codomain_shape = codomain_grid_shape
+        self.codomain_grid2world = codomain_grid2world
+
+    def set_affine(self, affine):
```

Comment:
Actually, maybe not, because you also modify the inverse, breaking the https://github.com/nipy/nibabel/wiki/property_manifesto